chain/chain/src/chain.rs

```diff
@@ -1288,6 +1345,8 @@ impl Chain {
     ) -> Result<Option<Tip>, Error> {
         metrics::BLOCK_PROCESSING_ATTEMPTS_TOTAL.inc();
         metrics::NUM_ORPHANS.set(self.orphans.len() as i64);
+        let block_hash = *block.hash();
+        let _timer = CryptoHashTimer::new(block_hash);
```

Comment:
This timer may not generate the time that you want. See how 'success_timer' is done here. `process_block` may return early because the block is orphan, or missing chunks, or the block is already known.

Reply:
yes, but the 'final' run should be the successful one, right? 
 (as without it, the block would not be added to the chain)

Reply:
Not necessarily. It's possible that the node receives a block that's already processed and `process_block` will return early because the block is known (see `check_known`)